### ISSUE TITLE
Automate release checklist tasks

### ISSUE THREAD

**Issue by jodh-intel** [01org/cc-oci-runtime#851]
Ideally, we'd have a script that could run the steps listed on the wiki here:

- https://github.com/01org/cc-oci-runtime/wiki/Release-Checklist

We're getting close to being able to automatically handle the OBS aspects (#672), which then leaves us with the question of whether we can automate the generation of the release notes.

**Comment by dlespiau** [comment]
We can at least generate the list of fixed bugs by:

1.  parsing `git log` and looking for `Fixes:` tags.
2. query the list of issues fixed between 2 dates. The latter is more exhaustive if we forget to put `Fixes:` tags and will have false positives (issues we have closed but not fixed).

I'd vote for 1. and have everyone reminded about those Fixes: tags.

**Comment by jodh-intel** [comment]
Agreed. I think as reviewers we should consider start requiring atleast one `Fixes #` in the commit messages.

**Comment by jodh-intel** [comment]
@grahamwhaley, @sameo - what do you think? Should we require every PR to be coupled with an issue along with a `Fixes #` comment?

**Comment by grahamwhaley** [comment]
My thought was that, sure, auto-grabbing and filtering for 'fixes' and similar is good and fine - but, I also expect something will always slip through the net. I expect a final stage in the release process will always have to be a visual sanity check and xref to ensure the auto generated notes container all the right things (and none of the 'wrong things'). Automation good - but I don't think we can trust it 100% to generate the final notes without human review.

**Comment by jodh-intel** [comment]
absolutely +1.

**Comment by devimc** [comment]
```
- but I don't think we can trust it 100% to generate the final notes without human review.
```
agree, we can't trust

**Comment by sameo** [comment]
As we enter into a more stable phase for COR, we do want PRs to be coupled with at least one issue. So I agree we should enforce the `Fixes` tag in PRs. We could even have pullapprove verify that for us.

**Comment by jodh-intel** [comment]
I've raised #856 to update our contribution guidelines. See the same issue for details of pullapprove.
